Documentation/gettingstarted/external-workloads.rst

```diff
@@ -0,0 +1,353 @@
+
+    ``<node-ip>`` is the node IP you extracted from the k8s cluster above.
+
+  .. group-tab:: LoadBalancer
+
+    .. parsed-literal ::
+
+      CLUSTER_ADDR=<load-balancer-ip> CILIUM_IMAGE=cilium/cilium:|IMAGE_TAG| ./install-external-workload.sh
+
+    ``<load-balancer-ip>`` is the load balancer IP you extracted from the k8s cluster above.
+
+This command launches the Cilium agent in a docker container named
+``cilium`` and copies the ``cilium`` CLI to your host. This needs
+``sudo`` permissions, so you may be asked for a password.
+
+This command waits until the node has been connected to the cluster
+and the cluster services are available. Then it re-configures
+``/etc/resolv.conf`` with the IP address of the ``kube-dns`` service.
+
+.. note::
+
+    If your node has multiple IP addresses you may need to tell Cilium
+    agent which to use. To this end add ``HOST_IP=<ip-address>`` to
+    the beginning of the command line above.
+
+Verify basic connectivity
+=========================
+
+Next you can check the status of the Cilium agent in your external workload:
+
+.. code:: bash
+
+    cilium status
+
+You should see something like:
+
+.. parsed-literal::
+
+    etcd: 1/1 connected, lease-ID=7c02748328e75f57, lock lease-ID=7c02748328e75f59, has-quorum=true: 192.168.36.11:32379 - 3.4.13 (Leader)
+    ...
+
+Check that cluster DNS works:
+
+.. code:: bash
+
+    nslookup clustermesh-apiserver.kube-system.svc.cluster.local
+
+Inspecting status changes in the cluster
+========================================
+
+The following command in your cluster should show the external workload IPs and their Cilium security IDs:
+
+.. code:: bash
+
+    kubectl get cew
+
+External workloads should also be visible as Cilium Endpoints:
+
+.. code:: bash
+
+    kubectl get cep
+
+Apply Cilium Network Policy to enforce traffic from external workloads
+======================================================================
+
+Ping the backend IP of ``clustermesh-apiserver`` service to verify connectivity:
+
+.. code:: bash
+
+    ping $(cilium service list get -o jsonpath='{[?(@.spec.flags.name=="clustermesh-apiserver")].spec.backend-addresses[0].ip}')
+
+The ping should keep running also when the following CCNP is applied in your cluster:
+
+.. parsed-literal::
+
+    apiVersion: cilium.io/v2
+    kind: CiliumClusterwideNetworkPolicy
+    metadata:
+      name: test-ccnp
+      namespace: kube-system
+    spec:
+      endpointSelector:
+        matchLabels:
+          k8s-app: clustermesh-apiserver
+      ingress:
+      - fromEndpoints:
+        - matchLabels:
+            io.kubernetes.pod.name: runtime
+      - toPorts:
+        - ports:
+          - port: "2379"
+            protocol: TCP
+
+The ping should stop if you delete these lines from the policy (e.g., ``kubectl edit ccnp test-ccnp``):
+
+.. parsed-literal::
+
+      - fromEndpoints:
+        - matchLabels:
+            io.kubernetes.pod.name: runtime
```

Comment:
While I think this does technically prove that security enforcement operates with the external workload being handled via a rich Identity, IMO it doesn't demonstrate the intended use cases for external workloads very well (eg, allowing access towards an external database, etc. I wonder if we could come up with a better example.

This guide also stops abruptly after this, it would be nice to have some summary text to establish what has now happened and what is possible.

Reply:
Added clean-up and conclustions sections.